### PR TITLE
BROCCOLI

### PR DESCRIPTION
**Pull request by wanderine**
Added wrappers for 3 BROCCOLI functions

Here is an example of how to use the functions

https://github.com/wanderine/BROCCOLI/blob/master/examples/nipypetest.py

### CODE REVIEW

**Comment by satra** [comment]
@wanderine - could you please merge with current master? there are a few things in the setup that have changed. otherwise, this will be a fun set of additions to play with.

also you could add a few tests to run_examples and use circle ci to test the broccoli interfaces as well.

once you are done writing the interfaces you should do a `make check-before-commit` to ensure that the auto tests and the docs are still building. the make command may add new files to your directory, so merging with upstream master is important before doing the make.

**Comment by chrisfilo** [comment]
I'm not sure CircleCI provides any GPU we can test on...

**Comment by wanderine** [comment]
BROCCOLI runs on CPUs as well.

**Comment by chrisfilo** [comment]
Oh - I think they have some of those ;)

**Comment by wanderine** [comment]
I don't understand this error

File "/home/andek/Research_projects/nipype/nipype/interfaces/broccoli/preprocess.py", line 71, in nipype.interfaces.broccoli.preprocess.RegisterTwoVolumes
Failed example:
    reg.cmdline
Expected:
    'RegisterTwoVolumes structural.nii mni.nii -platform 1 -device 2'
Got:
    'RegisterTwoVolumes structural.nii mni.nii -device 2 -platform 1'

File: nipype/interfaces/broccoli/preprocess.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    >>> from nipype.interfaces import broccoli
+    >>> from nipype.testing import example_data
+    >>> reg = broccoli.RegisterTwoVolumes()
+    >>> reg.inputs.in_file = 'structural.nii'
+    >>> reg.inputs.reference = 'mni.nii'
+    >>> reg.inputs.platform = 1
+    >>> reg.inputs.device = 2
+    >>> reg.cmdline
+    'RegisterTwoVolumes structural.nii mni.nii -platform 1 -device 2'
```
Comment (oesteban) [inline nipype/interfaces/broccoli/preprocess.py]: This doctest expects ```'RegisterTwoVolumes structural.nii mni.nii -device 2 -platform 1'```

**Comment by oesteban** [comment]
If ordering of -platform and -device is not important, then the change I suggest should fix the problem.

**Comment by wanderine** [comment]
OK, thanks.

CircleCI needs an OpenCL driver if I'm going to test BROCCOLI on it.

**Comment by chrisfilo** [comment]
You can add installation of required dependencies in the circle.yml file.

**Comment by wanderine** [comment]
What is the  operating system on CircleCI?

Does it have Intel or AMD CPU(s) ?

**Comment by chrisfilo** [comment]
Circle CI runs Ubuntu AFAIK. It's using EC2 so probably Intel but no
guarantees.

**Pull request closed without merging**